src/dragonball/src/device_manager/console_manager.rs

```diff
@@ -0,0 +1,427 @@
+// Copyright (C) 2022 Alibaba Cloud. All rights reserved.
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Portions Copyright 2017 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the THIRD-PARTY file.
+
+//! Virtual machine console device manager.
+//!
+//! A virtual console are composed up of two parts: frontend in virtual machine and backend in
+//! host OS. A frontend may be serial port, virtio-console etc, a backend may be stdio or Unix
+//! domain socket. The manager connects the frontend with the backend.
+//!
+//! Note: unwrap() is used without documentation, it's OK here.
+use std::io::{self, Read};
+use std::os::unix::net::{UnixListener, UnixStream};
+use std::path::Path;
+use std::sync::{Arc, Mutex};
+
+use bytes::{BufMut, BytesMut};
+use dbs_legacy_devices::{ConsoleHandler, SerialDevice};
+use dbs_utils::epoll_manager::{
+    EpollManager, EventOps, EventSet, Events, MutEventSubscriber, SubscriberId,
+};
+use vmm_sys_util::terminal::Terminal;
+
+use super::{DeviceMgrError, Result};
+
+const EPOLL_EVENT_SERIAL: u32 = 0;
+const EPOLL_EVENT_SERIAL_DATA: u32 = 1;
+const EPOLL_EVENT_STDIN: u32 = 2;
```

Comment:
Could these be an `enum` instead?

Reply:
Since we have 
```
 match slot {
            EPOLL_EVENT_SERIAL => {
                if let Err(e) = self.uds_listener_accept(ops) {
                    slog::warn!(self.logger, "failed to accept incoming connection, {:?}", e);
                }
            }
            EPOLL_EVENT_SERIAL_DATA => {
                if let Err(e) = self.uds_read_in(ops) {
                    slog::warn!(self.logger, "failed to read data from UDS, {:?}", e);
                }
            }
            EPOLL_EVENT_STDIN => {
                if let Err(e) = self.stdio_read_in(ops) {
                    slog::warn!(self.logger, "failed to read data from stdin, {:?}", e);
                }
            }
```
it will be kinda complex if we match slot(u32) with the enum